source/common/network/utility.cc

```diff
@@ -229,17 +229,19 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
 
     const Api::SysCallIntResult rc =
         Api::OsSysCallsSingleton::get().getifaddrs(interface_addresses);
-    RELEASE_ASSERT(!rc.return_value_, fmt::format("getifaddrs error: {}", rc.errno_));
-
-    // man getifaddrs(3)
-    for (const auto& interface_address : interface_addresses) {
-      if (!isLoopbackAddress(*interface_address.interface_addr_) &&
-          interface_address.interface_addr_->ip()->version() == version) {
-        ret = interface_address.interface_addr_;
-        if (ret->ip()->version() == Address::IpVersion::v6) {
-          ret = ret->ip()->ipv6()->addressWithoutScopeId();
+    if (!rc.return_value_) {
```

Comment:
I think the return_value should be 0 when getifaddrs succeeds?
In that case this should probably be `if (rc.return_value_) {`

Reply:
Oh, good point. That's embarrassing. (I'm surprised no tests failed. That's a bit weird) Switched to " != 0" to be explicit.